airbyte-integrations/connectors/source-prestashop/source_prestashop/components.py

```diff
@@ -0,0 +1,126 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+import datetime
+from dataclasses import InitVar, dataclass, field
+from typing import Any, ClassVar, Iterable, List, Mapping, Optional, Union
+
+import requests
+from airbyte_cdk.models import SyncMode
+from airbyte_cdk.sources.declarative.interpolation.interpolated_string import InterpolatedString
+from airbyte_cdk.sources.declarative.requesters.paginators.paginator import Paginator
+from airbyte_cdk.sources.declarative.requesters.paginators.strategies.pagination_strategy import PaginationStrategy
+from airbyte_cdk.sources.declarative.requesters.request_option import RequestOption
+from airbyte_cdk.sources.declarative.stream_slicers.stream_slicer import StreamSlicer
+from airbyte_cdk.sources.declarative.types import Config, Record, StreamSlice, StreamState
+from dataclasses_jsonschema import JsonSchemaMixin
+
+
+@dataclass
+class LimitOffsetPaginator(Paginator, JsonSchemaMixin):
+
+    limit_offset_option: RequestOption
+    pagination_strategy: PaginationStrategy
+    options: InitVar[Mapping[str, Any]]
+    _token: Optional[Any] = field(init=False, repr=False, default=None)
+
+    def next_page_token(self, response: requests.Response, last_records: List[Mapping[str, Any]]) -> Optional[Mapping[str, Any]]:
+        self._token = self.pagination_strategy.next_page_token(response, last_records)
+        if self._token:
+            return {"next_page_token": self._token}
+        else:
+            return None
+
+    def path(self) -> Optional[str]:
+        return None
+
+    def get_request_params(
+        self,
+        *,
+        stream_state: Optional[StreamState] = None,
+        stream_slice: Optional[StreamSlice] = None,
+        next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Mapping[str, Any]:
+        page_size = self.pagination_strategy.get_page_size()
+        if self._token:
+            return {self.limit_offset_option.field_name: f"{self._token},{page_size}"}
+        return {self.limit_offset_option.field_name: str(page_size)}
+
+    def get_request_headers(self, **kwargs) -> Mapping[str, str]:
+        return {}
+
+    def get_request_body_data(self, **kwargs) -> Mapping[str, Any]:
+        return {}
+
+    def get_request_body_json(self, **kwargs) -> Mapping[str, Any]:
+        return {}
+
+    def reset(self):
+        self.pagination_strategy.reset()
+
+
+@dataclass
+class NoSlicer(StreamSlicer, JsonSchemaMixin):
```

Comment:
why don't we want slices? The absence of slicing makes it impossible to checkpoint long syncs

Reply:
As for me there is no reason to add date-range slicing to this connector.
All streams can be simply iterated with pagining.
Because all stream are sorted by `coursor_field` I have added `checkpoint_interval: 500` to all incremental streams

Reply:
I'd like to better understand the benefit of using this custom implementation and your concerns with using stream slices.

If the only concern is that the connector might submit extra requests for date ranges with no data, can we enhance the `DatetimeStreamSlicer` slicers so it supports single slices instead of reimplementing most of the logic here?

Reply:
@girarda I have re-implemented using `DatetimeStreamSlicer`